elastic/kibana
# [Fleet] Fix navigation to policy for policies created inline

## Summary

Fixes https://github.com/elastic/kibana/issues/122157

When creating an inline policy, we were navigating back to whatever policy was selected in the dropdown - not the new policy created. This PR adds a check for whether we created a new policy and need to navigate to that policy instead.

## How to test

1. Navigate to any agent policy
2. Select "Add Integration"
3. Add any integration, e.g. Endpoint
4. In the "Apply to agent policy" section, select "Create agent policy"
5. Create a new agent policy
6. Submit the form
7. In the confirmation modal, select "Add agent to your hosts"
8. Observe that you are navigated to the new agent policy's details page
